__tests__/client/containers/edit-organization.js

```diff
@@ -1,24 +1,139 @@
 /* global describe, expect, it */
 
-import { mount } from 'enzyme'
-import { mountToJson } from 'enzyme-to-json'
+import {mount} from 'enzyme'
+import {mountToJson} from 'enzyme-to-json'
 import React from 'react'
-import { Provider } from 'react-redux'
+import {Provider} from 'react-redux'
 
-import { makeMockStore, mockStoreData } from '../../../test-utils/mock-store.js'
+import {makeGenericModelActionsExpectations} from '../../test-utils/actions'
+import {blankOrganization, makeMockStore, mockStores} from '../../test-utils/mock-data.js'
 
 import EditOrganization from '../../../client/containers/edit-organization'
 
-const mockStore = makeMockStore(mockStoreData)
+const organizationExpectations = makeGenericModelActionsExpectations({
+  pluralName: 'organizations',
+  singularName: 'organization'
+})
 
 describe('Container > EditOrganization', () => {
-  it('renders correctly', () => {
+  it('Create/Edit Organization View loads (create or edit mode)', () => {
+    const mockStore = makeMockStore(mockStores.withBlankAgency)
+
+    // mount component
+    const tree = mount(
+      <Provider store={mockStore}>
+        <EditOrganization
+          params={{agencyId: 'agency-1'}}
+          />
+      </Provider>
+    )
+    expect(mountToJson(tree.find(EditOrganization))).toMatchSnapshot()
+  })
+
+  it('Create/Edit Organization View loads in edit mode', () => {
+    const mockStore = makeMockStore(mockStores.withBlankOrganization)
+
     // mount component
     const tree = mount(
       <Provider store={mockStore}>
-        <EditOrganization />
+        <EditOrganization
+          params={{organizationId: 'organization-1'}}
+          />
       </Provider>
     )
     expect(mountToJson(tree.find(EditOrganization))).toMatchSnapshot()
   })
+
+  it('Create organization', () => {
+    const mockStore = makeMockStore(mockStores.withBlankAgency)
+
+    // mount component
+    const tree = mount(
+      <Provider store={mockStore}>
+        <EditOrganization
+          params={{agencyId: 'agency-1'}}
+          />
+      </Provider>
+    )
+
+    // give each text field some input
+    tree.find('input').map((input) => input.simulate('change', {target: {value: 'My new value'}}))
+
+    // submit form
+    tree.find('form').find('button').simulate('click')
+
+    // expect create action
+    organizationExpectations.expectCreateAction({
```

Comment:
There are tradeoffs to re-using the actions in the store.  The container tests have sort of become an integration test of container, component and action all at once.  If one were to send in mock functions and test the component directly that way it would truly become a unit test.  However, I do like the integration-style test since I can make sure all these things work together.